pkgs/checks/test/extensions/async_test.dart

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:async/async.dart';
+import 'package:checks/checks.dart';
+import 'package:test/scaffolding.dart';
+
+void main() {
+  group('FutureChecks', () {
+    test('completes', () async {
+      (await checkThat(_futureSuccess()).completes()).equals(42);
+
+      // TODO: validate the failure case
```

Comment:
Hmm, we might need an async version of `softCheck` for this.

Reply:
Or perhaps we do something like

```dart
test('completes - fails for an error future', () async {
  Future<void> failingTest() async {
    await checkThat(_futureFail()).completes();
  }
  await checkThat(failingTest()).throw();
}
```

Reply:
All we get is a `TestFailure` exception – can't validate the guts, which I think we want!

Reply:
Yeah, having the `Rejection` would be nicer. I thought that async `softCheck` would be challenging, but it turns out not to be.

https://github.com/dart-lang/test/pull/1789